scripts/discourse-sync/client.py

```diff
@@ -0,0 +1,22 @@
+import pydiscourse
+
+# Client extends pydiscourse.DiscourseClient with new methods
+class Client(pydiscourse.DiscourseClient):
+    def lock_post(self, post_id, locked, **kwargs):
+        """
+        Lock a post from being edited
+        https://docs.discourse.org/#tag/Posts/operation/lockPost
+
+        Args:
+            post_id: integer
+            locked: bool
+        """
+        if bool(locked):
+            kwargs["locked"] = "true"
+        else:
+            kwargs["locked"] = "false"
+        return self._put("posts/{}/locked.json".format(post_id), **kwargs)
```

Comment:
Assuming this is Python 3.6 or higher, you can use an f-string here. This is mostly style, but is generally what people would do these days:

```suggestion
        return self._put(f"posts/{post_id}/locked.json", **kwargs)
```

The way you're putting `locked` into `kwargs` should work. I think I would probably do it like this, though:

```suggestion
        return self._put(f"posts/{post_id}/locked.json", locked="true" if locked else false, **kwargs)
```

(Removing the `if` statement above entirely - ie. the method can just be one line.